feat: add the pull based fileserver (#1284)
celestiaorg/celestia-core: pkg/trace/README.md

directory.

```toml
# The tracer url token to use for remote event collection.
# If empty, remote event pulling is disabled.

> trace_pull_address = ":26661"

Reviewer: TBH I don't understand what "url token" means here. The value looks like a port. Can the comment be made more clear?

If the config is supposed to be a value, why is the address portion omitted and only the port there?

Author: that was just old docs, but good catch fixed!

# comma separate string. For example: "consensus_round_state,mempool_tx".
tracing_tables = "consensus_round_state,mempool_tx"
```

To retrieve a table remotely using the pull based server, call the following
function:

```go
err := GetTable("http://1.2.3.4:26661", "mempool_tx", "directory to store the file")
if err != nil {
    return err
}
```

This will download and save the file to the specified directory in the

> `table_name.jsonl` format.

Reviewer: [no change needed] TIL there is a JSONL format. Ref: https://jsonlines.org/

Author: it;s pretty neat

> To read the contents of the file, load it and pass it the Decode function.

Reviewer: ```suggestion
To read the contents of the file, invoke the `DecodeFile` function.
```
